solidity/contracts/token/HypValue.sol

```diff
@@ -0,0 +1,166 @@
+     */
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount,
+        bytes calldata _hookMetadata,
+        address _hook
+    ) external payable virtual override returns (bytes32 messageId) {
+        uint256 quote = _checkSufficientValue(_destination, _amount, _hook);
+
+        bytes memory hookMetadata = StandardHookMetadata.overrideMsgValue(
+            _hookMetadata,
+            _amount
+        );
+
+        return
+            _transferRemote(
+                _destination,
+                _recipient,
+                _amount,
+                _amount + quote,
+                hookMetadata,
+                _hook
+            );
+    }
+
+    /// @inheritdoc TokenRouter
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount
+    ) external payable virtual override returns (bytes32 messageId) {
+        uint256 quote = _checkSufficientValue(
+            _destination,
+            _amount,
+            address(hook)
+        );
+        bytes memory hookMetadata = StandardHookMetadata.formatMetadata(
+            _amount,
+            destinationGas[_destination],
+            msg.sender,
+            ""
+        );
+
+        return
+            _transferRemote(
+                _destination,
+                _recipient,
+                _amount,
+                _amount + quote,
+                hookMetadata,
+                address(hook)
+            );
+    }
+
+    // ============ Internal Functions ============
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev No token metadata is needed for value transfers
+     */
+    function _transferFromSender(
+        uint256
+    ) internal pure override returns (bytes memory) {
+        return bytes(""); // no token metadata
+    }
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev Sends the value to the recipient
+     */
+    function _transferTo(
+        address _recipient,
+        uint256 _amount,
+        bytes calldata // no token metadata
+    ) internal virtual override {
+        Address.sendValue(payable(_recipient), _amount);
+    }
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev This contract doesn't hold value
+     */
+    function balanceOf(
+        address /* _account */
+    ) external pure override returns (uint256) {
+        return 0;
+    }
+
+    /// @dev Checks if the provided value is sufficient for the transfer
+    function _checkSufficientValue(
+        uint32 _destination,
+        uint256 _amount,
+        address _hook
+    ) internal view returns (uint256) {
+        uint256 quote = _GasRouter_quoteDispatch(
+            _destination,
+            new bytes(0),
+            _hook
+        );
+        if (msg.value < _amount + quote) {
+            revert InsufficientValue(_amount + quote, msg.value);
+        }
+        return quote;
```

Comment:
its odd to me that you return quote here and then also do `amount + quote` outside of this function too

Reply:
fixed